### PR TITLE
feat(webpack): this adds some ezpz externals and whitelisting support for webpacking npmcli

### PR DESCRIPTION
**Pull request by TheLarkInn**
This commit does the following: 

* Ignores _truly dynamic requires_ by wrapping them in `new Function`
* Expands the reach of `shebang-loader` (therefore any js file with the env/shebang can be consumed)
* adds webpack-node-externals and adds the "bundledDependencies" list to the whitelist with a few hacked on dependencies missing (yargs, update-notifier, etc.)
* makes a local "dynamic" require a context module

### CODE REVIEW

File: scripts/webpack-bundle.js [hunk 1]
```diff
@@ -40,6 +42,128 @@ function bundler (opts) {
...
+          "which",
+          "wrappy",
+          "write-file-atomic",
+          "safe-buffer",
+          "tiny-relative-date",
+          "cli-columns",
+          "node-gyp"
+        ]
+      })
+    ],
```
Comment (zkat) [inline scripts/webpack-bundle.js]: Can you make this `require('../package.json').bundleDependencies.filter(...)` so we know which specific packages are being blacklisted?

File: lib/npm.js [hunk 2]
```diff
@@ -124,7 +124,7 @@
       npm.command = c
       if (commandCache[a]) return commandCache[a]
 
-      var cmd = require(path.join(__dirname, a + '.js'))
+      var cmd = require("./" + path.join(__dirname, a) + ".js");
```
Comment (zkat) [inline lib/npm.js]: This needs to be `''` and drop the `;`, because `standard`.

File: scripts/webpack-bundle.js [hunk 3]
```diff
@@ -19,7 +21,7 @@ if (require.main === module) {
 }
 
 module.exports = bundler
-function bundler (opts) {
+function bundler(opts) {
```
Comment (zkat) [inline scripts/webpack-bundle.js]: Put this space back, because `standard`.

File: scripts/webpack-bundle.js [hunk 4]
```diff
@@ -31,7 +33,7 @@ function bundler (opts) {
     node: false,
     module: {
       rules: [
-        {test: /np.-cli.js$/, loader: 'shebang-loader'}
+        { test: /.js$/, loader: 'shebang-loader' }
```
Comment (zkat) [inline scripts/webpack-bundle.js]: I guess this is probably reasonable tbh.

File: scripts/webpack-bundle.js (hunk 1 above)
Comment (zkat) [reply scripts/webpack-bundle.js]: Also, does this track transitive dependencies?

**Review by zkat — CHANGES_REQUESTED** [review]

File: lib/npm.js (hunk 2 above)
Comment (zkat) [reply lib/npm.js]: Oh and this needs to be `'./' + a + '.js'`. Drop the __dirname.

Comment (TheLarkInn) [reply lib/npm.js]: perfect!! Will change.

File: scripts/webpack-bundle.js (hunk 1 above)
Comment (TheLarkInn) [reply scripts/webpack-bundle.js]: Oh yeah thats a great idea! I'll do so.

**Comment by zkat** [comment]
wait what why did it close D:

**Comment by zkat** [comment]
I guess it got merged. Heh.

**Pull request closed without merging**